src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -330,25 +330,39 @@ public Uri ResolveOperationDefinitionUrl(string operationName)
 
         private Uri GetRouteUri(HttpContext httpContext, string routeName, RouteValueDictionary routeValues, string scheme, string host)
         {
-            try
+            var s = string.Empty;
+            routeValues?.ToList().ForEach(kv => s += kv.ToString());
+
+            var uriString = string.Empty;
+            if (httpContext == null)
             {
-                var uriString = UrlHelper.RouteUrl(
+                // Keep it for UTs
+                uriString = UrlHelper.RouteUrl(
                     routeName,
                     routeValues,
                     scheme,
                     host);
-                return new Uri(uriString);
             }
-            catch (InvalidOperationException)
+            else
             {
-                var uriString = _linkGenerator.GetUriByRouteValues(
+                // NOTE: Append "/" to the end of the path base to workaround a known bug in UrlHelper.RouteUrl for endpoint routing.
+                //       Remove the workaround when we pick up the fix. (https://github.com/dotnet/aspnetcore/issues/53177)
```

Comment:
We should create a user story to remove this workaround

Reply:
Created user story here - https://microsofthealth.visualstudio.com/Health/_workitems/edit/113806